### PR TITLE
Add an 'atomics' watcher type that uses the BEAM atomic functions

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

**Comment by ferd** [comment]
Gonna need to rebase to fix the conflicts. It appears fine otherwise, but I'll give it a deeper look after the conflict is fixed.

**Comment by Vagabond** [comment]
Rebased.

File: test/prop_dispcount.erl [hunk 1]
```diff
@@ -75,7 +76,26 @@ prop_nocrash() ->
...
+    prop_nocrash(named, round_robin).
+
+prop_nocrash_atomics_hash() ->
+    prop_nocrash(atomics, hash).
+
+prop_nocrash_atomics_round_robin() ->
+    prop_nocrash(atomics, round_robin).
+
+%% this seems broken?
+noprop_parallel_nocrash() ->
```
Comment (ferd) [inline test/prop_dispcount.erl]: possibly? I don't think I truly understood stateful tests back then. Seems to work as a suite though.

**Comment by ferd** [comment]
let me know if you need a version bump. I can probably update the config files as well.

File: test/prop_dispcount.erl (hunk 1 above)
Comment (Vagabond) [reply test/prop_dispcount.erl]: I couldn't figure out how to run it in a way that didn't make it fail, pointers welcome.

**Pull request merged**